src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1809,43 +1822,61 @@ applyTransaction (BasicView& view,
     }
 }
 
-void applyTransactions (
-    SHAMap const* set,
+// Build a transaction set from a transaction map.
+// This allows us to deserialize the transactions only once.
+TxSet buildTxSet(SHAMap const& set)
```

Comment:
You could return a transformed boost::range using SHAMap::iterator
```
boost::adaptors::transform(
    set, [](SHAMapItem const& item)
    {
        return std::make_shared<STTx const>(
            SerialIter{item->slice()});
    }
```

This replaces the `buildTxSet` function. The `apply` routine would have to be templated.